_drafts/oryx/ssr.md

```diff
@@ -0,0 +1,146 @@
+    component: '<root-app></root-app>',
+});
+```
+
+### Caching, CDN
+
+Caching and CDNs are additional layers that can significantly enhance the performance of SSR applications by reducing server load and accelerating content delivery.
+
+**Caching**: Caching acts as a layer on top of rendering. Once the HTML content is rendered, it's cached to swiftly serve repeated requests without the need for re-rendering. Various caching solutions could be used, such as Varnish, Redis, or even service-specific solutions such as Cloudflare's caching services.
+
+**Content Delivery Networks (CDNs)**: CDNs distribute cached content across a network of servers located worldwide. This ensures that users receive content from the nearest server, which significantly improves delivery speed and reduces latency.
+
+In the context of serverless platforms like Netlify, you can leverage Netlify's on-demand builder.
+It works as a simple wrapper to SSR Lambda handler and automatically feed Netlify Edge CDN by building and caching a page when it's requested, thereby reducing the load on the serverless function and improving performance.
+
+```mermaid
+sequenceDiagram
+  participant User
+  participant CDN
+  participant SSR Service
+  participant Backend
+  
+  User->>CDN: Request page
+    CDN->>SSR Service: Request SSR page
+  CDN-->>User: Deliver SSR page
+  User->>CDN: Request static assets (CSR)
+  CDN-->>User: Deliver static JS assets
+  Note over User: Hydration occurs, SPA mode
+  User->>Backend: CSR Interacts with backend app
+```
+
+## SSR in Oryx
+
+### Configuration
+
+Oryx integrates seamlessly with server environments, offering support for both traditional Node.js server-side rendering (SSR) and serverless architectures, akin to AWS Lambda or Netlify functions.
+
+For Node.js SSR, Oryx leverages ExpressJS, a minimalist web framework for Node.js. By utilizing the `createServer` method, Oryx is capable of spinning up an ExpressJS server configured for SSR out of the box.
+
+Simultaneously, Oryx caters to the increasing demand for serverless architectures. The `storefrontHandler` method enables SSR in AWS Lambda-like environments, providing a flexible solution for serverless deployments.
+
+While both approaches offer sensible configuration tailored to most deployment scenarios, it also exposes a lower-level API, allowing advanced customization to cater to unique project requirements.
+
+### Build with SSR support
+
+Oryx has been designed from the ground up with SSR support in mind, ensuring that all components render correctly on the server side. This ingrained support for SSR enables Oryx to deliver faster initial page load times and improved SEO out of the box.
+
+Moreover, Oryx supports late and partial hydration that further optimizes performance. This approach ensures that only the necessary components are hydrated when they are required, reducing the amount of JavaScript that needs to be parsed and executed on the client side.
+
+### 3.3 Hydration
```

Comment:
Maybe we can switch `Hydration` and `Build with SSR support` sections as long as we are discussing hydration in it, so it's logical to introduce a topic to the reader before we start mentioning it.

Reply:
improved a bit, by mentioning hydration later :)